model/histogram/float_histogram.go

```diff
@@ -352,7 +354,11 @@ func (h *FloatHistogram) Add(other *FloatHistogram) (*FloatHistogram, error) {
 		h.ZeroCount += otherZeroCount
 	}
 	h.Count += other.Count
-	h.Sum += other.Sum
+	// Kahan summation algorithm
+	y := other.Sum - h.KahanSumHint
+	t := h.Sum + y
+	h.KahanSumHint = (t - h.Sum) - y
+	h.Sum = t
```

Comment:
Note that there already is a function `kahanSumInc` in promql/engine.go. (Which cannot simply be used here because it would cause a circular package dependency. Let's discuss that later.)

More importantly: This only handles the `Sum` field, but Kahan summation should be used on _all_ the fields that get added up. I think we need a very different approach here.